packages/opentelemetry/src/utils/spanData.ts

```diff
@@ -7,7 +7,13 @@ import type { AbstractSpan } from '../types';
 // This way we can enhance the data that an OTEL Span natively gives us
 // and since we are using weakmaps, we do not need to clean up after ourselves
 const SpanScope = new WeakMap<AbstractSpan, Scope>();
-const SpanFinishScope = new WeakMap<AbstractSpan, Scope>();
+const SpanFinishScopes = new WeakMap<
+  AbstractSpan,
+  {
+    scope: Scope;
+    isolationScope: Scope;
+  }
+>();
```

Comment:
h: Do we really need this if the span has the scopes attached to it already? Also, how is this different from `SpanScope` and `setSpanScope`?

Reply:
the tricky thing here is that we use _this_ here to "cache" the scopes on the _OTEL_ spans, then we use `setCapturedScopes()` to set them on the _sentry_ span when we create it in the span exporter.

Once we have aligned our span types, we can probably use `setCapturedScopes` here as well - but we'll still need to pick them up in the span exporter via `getCapturedScopes` and set them again on the _sentry_ span before we send it.

Reply:
we can probably get rid of `getSpanScope` actually, I'll check it out!